plugins/python/parser/src/scripts/cc_python_parser/import_preprocessor.py

```diff
@@ -0,0 +1,179 @@
+        return self.name == '*'
+
+
+class Import:
+    def __init__(self, path: List[str], module: ModuleImport, imported: List[SymbolImport],
+                 location: Optional[PurePath], r: Range, is_local: bool = False):
+        self.path: List[str] = path
+        self.module: ModuleImport = module
+        self.imported: List[SymbolImport] = imported
+        self.location: Optional[PurePath] = location
+        self.range = r
+        self.is_local = is_local
+
+    def is_module_import(self):
+        return len(self.imported) == 0
+
+
+# TODO: need locations of files, and check if modules are among them?
+# TODO: handle relative import:
+#   '.': from the package (parent directory) of the current module (from .'package' import 'module')
+#   from the __init__ (from . import 'symbol')
+#   '..': from the grandparent directory of the current module (and so on)
+class ImportTable:
+    def __init__(self):
+        self.modules: List[Import] = []
+        self.import_paths: Set[PurePath] = set()
+
+    def append_import(self, node: ast.Import, is_local: bool = False):
+        self.modules.extend(self.convert_ast_import_to_import(node, is_local))
+        for module in node.names:
+            m = None
+            try:
+                if module.name != '__main__':
+                    m = util.find_spec(module.name)
+            except (AttributeError, ModuleNotFoundError, ImportError, ValueError):
+                pass
+            if m is not None and m.origin is not None:
+                self.import_paths.add(PurePath(m.origin))
+
+    def append_import_from(self, node: ast.ImportFrom, is_local: bool = False):
+        self.modules.extend(self.convert_ast_import_from_to_import(node, is_local))
+        m = None
+        try:
+            if node.level > 0 and node.module is None:
+                m = util.find_spec(node.names[0].name)
+            elif node.level == 0 or node.module is not None:
+                m = util.find_spec(node.module + '.' + node.names[0].name)
+            else:
+                pass    # print()
+        except (AttributeError, ModuleNotFoundError, ImportError, ValueError):
+            pass
+        except KeyError:
+            return
+        finally:
+            if m is None:
+                try:
+                    if node.module != '__main__':
+                        m = util.find_spec(node.module)
+                except (AttributeError, ModuleNotFoundError, ImportError, ValueError):
+                    pass
+                except KeyError:
+                    return
+            if m is not None and m.origin is not None:
+                self.import_paths.add(PurePath(m.origin))
+
+    @staticmethod
+    def convert_ast_import_to_import(node: ast.Import, is_local: bool = False) -> List[Import]:
+        imports = []
+        for module in node.names:
+            module_parts = module.name.split('.')
+            module_name = module_parts[-1]
+            module_parts.remove(module_name)
```

Comment:
```suggestion
            del module_parts[-1]
```
If the module name appears several times in the full module name then the first occurrence would be removed.